packages/esm-patient-forms-app/src/forms/forms-list.component.tsx

```diff
@@ -1,81 +1,69 @@
-  DataTableSkeleton,
-  Layer,
-  Link,
-  Table,
-  TableBody,
-  TableCell,
-  TableContainer,
-  TableHead,
-  TableHeader,
-  TableRow,
-  TableToolbar,
-  TableToolbarContent,
-  TableToolbarSearch,
-  Tile,
-} from '@carbon/react';
-import { formatDatetime, useLayoutType, Visit } from '@openmrs/esm-framework';
-import { EmptyDataIllustration, closeWorkspace } from '@openmrs/esm-patient-common-lib';
-import { HtmlFormEntryForm } from '../config-schema';
-import { launchFormEntryOrHtmlForms } from '../form-entry-interop';
-import { useForms } from '../hooks/use-forms';
-import styles from './forms-list.scss';
 import debounce from 'lodash-es/debounce';
+import fuzzy from 'fuzzy';
+import { DataTableSkeleton, Layer, Tile } from '@carbon/react';
+import { formatDatetime, useLayoutType } from '@openmrs/esm-framework';
+import { EmptyDataIllustration } from '@openmrs/esm-patient-common-lib';
+import FormsTable from './forms-table.component';
+import styles from './forms-list.scss';
+import type { CompletedFormInfo } from '../types';
 
 export type FormsListProps = {
-  currentVisit: Visit;
-  htmlFormEntryForms: Array<HtmlFormEntryForm>;
-  patient: fhir.Patient;
-  patientUuid: string;
+  completedForms?: Array<CompletedFormInfo>;
+  error?: any;
+  sectionName?: string;
+  handleFormOpen: (formUuid: string, encounterUuid: string, formName: string) => void;
 };
 
-const FormsList: React.FC<FormsListProps> = ({ currentVisit, htmlFormEntryForms, patient, patientUuid }) => {
+function ResponsiveWrapper({ children, isTablet }: { children: React.ReactNode; isTablet: boolean }) {
+  return isTablet ? <Layer>{children} </Layer> : <>{children}</>;
+}
+
+/*
+ * For the benefit of our automated translations:
+ * t('forms', 'Forms')
+ */
+
+const FormsList: React.FC<FormsListProps> = ({ completedForms, error, sectionName = 'forms', handleFormOpen }) => {
   const { t } = useTranslation();
-  const { data, error, mutateForms } = useForms(patientUuid);
   const [searchTerm, setSearchTerm] = useState('');
   const isTablet = useLayoutType() === 'tablet';
+  const [locale, setLocale] = useState(window.i18next.language ?? navigator.language);
+
+  useEffect(() => {
+    if (window.i18next?.on) {
+      const languageChanged = (lng: string) => setLocale(lng);
+      window.i18next.on('languageChanged', languageChanged);
+      return () => window.i18next.off('languageChanged', languageChanged);
+    }
+  }, []);
+
+  const handleSearch = useMemo(() => debounce((searchTerm) => setSearchTerm(searchTerm), 300), []);
 
   const filteredForms = useMemo(() => {
     if (!searchTerm) {
-      return data;
+      return completedForms;
     }
-    return data.filter((form) => {
-      const formName = form.form.display ?? form.form.name;
-      return formName.toLowerCase().includes(searchTerm.toLowerCase());
-    });
-  }, [data, searchTerm]);
 
-  const handleSearch = React.useMemo(() => debounce((searchTerm) => setSearchTerm(searchTerm), 300), []);
+    return fuzzy
```

Comment:
I like this fuzzy search approach. Perhaps we should use it in most other places.

Reply:
I got the idea from @brandones... I like that library a lot; it's simple and to the point. I'm not entirely sure how well it will perform with different languages and scripts, but it should be no worse than what was there before.